content/learn/beginner/control-flow-1.md

```diff
@@ -0,0 +1,213 @@
+we often need to make decisions based on certain conditions. This is where *control flow* comes in.
+
+You already have an intuitive understanding of control flow. For instance, when you cross the street, you look left and
+right to see if any cars are coming. If there are no cars, you cross the street. If there are cars, you wait until they
+have passed, then check again. This is a simple example of a decision-making process.
+
+## Comparison Actions
+
+In programming, we can make decisions by comparing values to each other. We can compare numbers, strings, and other data
+types. The result of a comparison is a *boolean* value, which is either `true` or `false`.
+
+{{< callout context="caution" title="Comparing across Types" icon="outline/alert-triangle" >}}
+
+Just like you cannot quite compare apples and oranges, you cannot compare values of different types. For instance, you
+cannot compare a number to a string. The bot will throw an error if you try to do so, you will have to convert either of
+them to the other type first.
+
+{{< /callout >}}
+
+We provide the following comparison functions in Custom Commands:
+
+- `eq` (equals `==`)
+- `ne` (not equals `!=`)
+- `lt` (less than `<`)
+- `le` (less than or equal to `<=`)
+- `gt` (greater than `>`)
+- `ge` (greater than or equal to `>=`)
+
+These functions can only compare basic data types as introduced in [Data Types 1](/learn/beginner/datatypes-1). Strings
+are compared using the [underlying Unicode values](https://en.wikipedia.org/wiki/List_of_Unicode_characters), stepping
+through the strings byte by byte. Please refer to the [functions documentation](/docs/reference/templates/functions)
+for the syntax of these functions.
+
+### If Statements
+
+The most basic form of control flow is the `if` statement. An `if` statement checks a condition and executes a block of
+code if the condition is true. If the condition is false, the block of code is skipped.
+
+Here is an example of an `if` statement:
+
+```go
+{{ if eq 5 5 }}
+  Five is equal to five!
+{{ end }}
+```
+
+We can expand this to execute a different block of code if the condition is false by using an `else` block:
+
+```go
+{{ if eq 5 3 }}
+  Five is equal to three!
+{{ else }}
+  Five is not equal to three!
+{{ end }}
+```
+
+This can be further expanded to check multiple conditions using `else if`, which are checked sequentially until one of
+them is true:
+
+```go
+{{ if eq 5 3 }}
+  Five is equal to three!
+{{ else if eq 5 5 }}
+  Five is equal to five!
+{{ else }}
+  Five is not equal to three or five!
+{{ end }}
+```
+
+#### Guard Clauses
+
+As your code grows, you may find yourself nesting `if` statements inside each other. This can lead to code that is hard
+to read and understand. One way to avoid this is to use *guard clauses*. A guard clause is an `if` statement that checks
+for a condition and returns early if the condition is false.
```

Comment:
We have not introduced `return` yet (I think?), so although the concept is simple it would still be nice to briefly elaborate: "and exits early using the `{{ return }}` action if the condition is false."

Reply:
You're quite right, we haven't introduced `return` yet. I think it's
sufficient enough to explain that we use `return` to exit early.
Later on, when we introduce `define` and friends, we can then expand the
definition of return and show how it's used to return a value from a
template.